vignettes/higher_order.Rmd

```diff
@@ -0,0 +1,205 @@
+---
+title: "Higher Order Functions"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Higher Order Functions}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+This vignette explains some of the more advanced options of `{admiral}` related
+to higher order functions. A higher order function is a function that takes another
+function as input. By introducing these higher order functions, we intend to give
+the user greater power over our derivation functions, whilst trying to negate the
+need for adding additional `{admiral}` functions or arguments, or the user needing
+many separate steps.
+
+The functions covered here are:
+* `call_derivation()`: Call a single derivation multiple times with some
+parameters/arguments being fixed across iterations and others varying
+* `restrict_derivation()`: Execute a derivation on a subset of the input dataset
+* `slice_derivation()`: The input dataset is split into slices (subsets) and for
+each slice the derivation is called separately. Some or all arguments of the
+derivation may vary depending on the slice.
+
+## Required Packages
+
+The examples of this vignette require the following packages.
+
+For example purpose, the CDISC Pilot SDTM datasets---which are included in
+`{admiral.test}`---are used.
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+library(admiral.test)
+library(dplyr)
+
+data("admiral_vs")
+data("admiral_ae")
+vs <- admiral_vs
+ae <- admiral_ae
+vs <- convert_blanks_to_na(vs)
+ae <- convert_blanks_to_na(ae)
+```
+```{r echo=FALSE}
+vs <- filter(vs, USUBJID %in% c("01-701-1015"))
+ae <- filter(ae, USUBJID %in% c("01-701-1111", "01-705-1393"))
+```
+
+# Call Derivation
+
+This function exists purely for convenience to save the user repeating numerous
+similar derivation function calls. It is best used when multiple derived variables
+have very similar specifications with only slight variations.
+
+As an example, imagine the case where all the parameters in a BDS ADaM required
+both a highest value flag and a lowest value flag.
+
+Here is an example of how to achieve this **WITHOUT** using `call_derivation()`: 
```

Comment:
I think bolding "without" is enough, no need for all uppercse. Same below for "with".

Reply:
ok